service/routes/dc.py

```diff
@@ -0,0 +1,57 @@
+import json
+from flask import Blueprint
+from flask import request
+from flask import Response
+from markupsafe import escape
+from ..utils import ApiUtils
+from ..utils import ReadFile
+from ..utils import InvalidUsage
+
+bp_dc = Blueprint("dc", __name__, url_prefix="/dc")
+
+
+@bp_dc.route("/", methods=["GET", "POST"])
+@bp_dc.route("/<characters>/", methods=["GET"])
+def dc(characters=None):
+    """
+    DC endpoint.
+    Flask defaults to trailing slash which works for both dc/badman and /dc/batman/
+    :param characters: (str) String representation of DC characters to search/filter by e.g batman
+    """
+    api = ApiUtils()
+    options = dict().copy()
+    options.update(request.args)
+
+    if request.method == "GET" and characters is not None:
```

Comment:
To run the tests -

cd simple-superhero-service-python
coverage run -m pytest

Reply:
So there is a reason why I've done this the way I have. After reading about it, it appears that the method-based-dispatching can't be intermixed with the blueprints (a.k.a application-based-dispatching). This is important because in the way that I've architected the service, I have loosely coupled classes which are easy to test and don't have inter-dependencies. The disadvantage is that there is a lot of overlap (repeated ideas/code) which I think is acceptable for 4 endpoints.

In your example above, it appears that the `view_func` from the `method-based-dispatching` is an all or nothing kind of deal. 

`app.add_url_rule('/dc/<characters>', view_func=DCAPI.as_view('characters'))`

If created, the DCAPI class would have inter-dependencies of the ReadFile class and the ApiUtils class which makes them tightly coupled and would be counter to what OOP principals are advocating. 

Perhaps I'm overthinking the OOP portion of this